lib/spack/spack/test/install.py

```diff
@@ -183,70 +183,65 @@ def test_flatten_deps(
     assert os.path.isdir(dependency_dir)
 
 
-def test_installed_upstream_external(
-        tmpdir_factory, install_mockery, mock_fetch, gen_mock_layout,
-        monkeypatch):
-    """Check that when a dependency package is recorded as installed in
-       an upstream database that it is not reinstalled.
-    """
+@pytest.fixture()
+def install_upstream(tmpdir_factory, gen_mock_layout, install_mockery):
```

Comment:
Could you add a docstring to this fixture? Something like:

> Provides a function that installs a specified set of specs to an upstream database. The function returns a store which points to the upstream, as well as the upstream layout (for verifying that dependent installs are using the upstream installs).

Reply:
Done in f3ceee6